src/js/overlay.js

```diff
@@ -225,16 +219,33 @@ Overlay.prototype.show = function() {
 
 	this.closeOnExternalClick = this.closeOnExternalClick.bind(this);
 	this.delegates.doc.on('click', 'body', this.closeOnExternalClick);
+	this.delegates.doc.on('touchstart', 'body', this.closeOnExternalClick);
```

Comment:
Aren't there ways to check if the touch is intentional? Not sure it's important here, though.

Also: are some devices going to trigger both click and touchstart? Should we trigger one before the other?